### PR TITLE
CSIT-913 BO steps and mobile view steps for Self Exclusion test

### PR DESCRIPTION
**Pull request by Alena-deriv**
Completed all steps for https://app.clickup.com/t/20696747/CSIT-913 (subtask https://app.clickup.com/t/20696747/CSIT-1120)
Changes made:
1. BO steps are added
2. Mobile view steps are added
3. Rearranged test steps to be logically divided (like /* Sets self exclusion on FE */)
4. Moved test from wip folder to smoke since all test cases are automated

Screenshot 2024-06-13 at 09 35 13

### CODE REVIEW

**Review by markodonnell-deriv — APPROVED** [review]
Looks good.

File: cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+describe('QATEST-116798 Self Exclusion Session and login limits on desktop', () => {
+  beforeEach(() => {
+    cy.c_visitResponsive('/')
+    cy.c_createRealAccount()
+    cy.c_login()
+  })
+
+  it('should login, set self exclusion, verify its applied, check from BO side, remove it from BO and verify restrictions are removed from FE side ', () => {
+    /* Sets self exclusion on FE */
+    cy.findByTestId('dt_traders_hub').should('be.visible') //temp solution
```
Comment (bilalmajid-deriv) [inline cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js:18]: Do we need this temp solution now?
@abuzar-deriv  can you confirm

**Review by bilalmajid-deriv — APPROVED** [review]

File: cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js (hunk 1 above)
Comment (abuzar-deriv) [reply cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js:18]: No longer needed.

Comment (Alena-deriv) [reply cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js]: This is removed

**Review by bilalmajid-deriv — APPROVED** [review]
LGTM

**Review by abuzar-deriv — CHANGES_REQUESTED** [review]
Why two different describe blocks for same test? Can be done with one describe, two its . Also I feel like a lot of these steps are duplicated such as /* Checks self exclusion is saved on FE */, /* Clears self exclusion date from BO */, and other similar areas which can be moved to a function and then called. 
Can you please review and move all the duplicate code into their separate functions?

File: cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+describe('QATEST-116798 Self Exclusion Session and login limits', () => {
+  beforeEach(() => {
+    cy.c_visitResponsive('/')
+    cy.c_createRealAccount()
+    cy.c_login()
+  })
+
+  size.forEach((size) => {
+    it('should login, set self exclusion, verify its applied, check from BO side, remove it from BO and verify restrictions are removed from FE side ', () => {
```
Comment (abuzar-deriv) [inline cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js:18]:
```suggestion
    it(`Should login, set self exclusion, verify it's applied, check from BO side, remove it from BO and verify restrictions are removed from FE side on {size=='small'?'mobile:'desktop'}`, () => {
```

**Review by abuzar-deriv — CHANGES_REQUESTED** [review]
One minor change needed.

File: cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js (hunk 2 above)
Comment (Alena-deriv) [reply cypress/e2e/smoke/CSIT/selfExclusionSessionAndLoginLimits.cy.js]: It is updated

**Review by abuzar-deriv — APPROVED** [review]
LGTM

**Review by meeyun-deriv — APPROVED** [review]
LGTM

**Pull request merged**